packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx

```diff
@@ -7,5 +7,7 @@ declare const ENTITY_TITLE_FIELD_NAME = "title";
 export const ENTITY_TITLE_FIELD_NAME_ID = ENTITY_TITLE_FIELD_NAME;
 
 export const ENTITY_TITLE = (record: ENTITY): string => {
-  return record.ENTITY_TITLE_FIELD || String(record.id);
+  return typeof record[ENTITY_TITLE_FIELD_NAME] === "string"
```

Comment:
you better change it to: Object.prototype.toString.call(str) === "[object String]"

Reply:
Done @barshimi.